src/main.cpp

```diff
@@ -302,11 +301,29 @@ void nimble_port_ll_task_func(void* args) {
 }
 }
 
+void calibrate_lf_clock_rc(nrf_drv_clock_evt_type_t event) {
+  // 16 * 0.25s = 4s calibration cycle
+  // Not recursive, call is deferred via internal calibration timer
+  nrf_drv_clock_calibration_start(16, calibrate_lf_clock_rc);
+}
+
```

Comment:
What's the purpose of this handler?

Reply:
Yes, see https://infocenter.nordicsemi.com/index.jsp?topic=%2Fcom.nordic.infocenter.sdk5.v15.0.0%2Fgroup__nrf__drv__clock.html . The RC clock drifts fast and source has to be re-calibrated every ~4 seconds, in order to stay accurate to the 500ppm (`MYNEWT_VAL_BLE_LL_SCA`) maximum allowed by the BLE stack.

The timer used for this loop is a dedicated hardware timer (`CT`), see https://infocenter.nordicsemi.com/index.jsp?topic=%2Fcom.nordic.infocenter.nrf52832.ps.v1.1%2Fclock.html . It does not interfere with any other timers.